Functional programming/Problem 1/partial_application.js

```diff
@@ -0,0 +1,15 @@
+function F() {
+	"use strict";
+
+	var toArray = Array.prototype.slice;
+	var argsCount = arguments.length - 1;
+	var args = toArray.call(arguments, 0, argsCount);
+	var g = arguments[argsCount];
```

Comment:
Supposition, that the last argument should be the function is a bad practice.  The more elegant way is to make it an implicit parameter: Function.prototype.partial = function